Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/ManageIISWebApp.ps1

```diff
@@ -0,0 +1,339 @@
+﻿Import-Module $env:CURRENT_TASK_ROOTDIR\DeploymentSDK\InvokeRemoteDeployment.ps1
+
+Write-Verbose "Entering script ManageIISWebApp.ps1"
+
+
+function Set-IISWebSite
+{
+    param (
+        [string] $actionIISWebsite,
+        [string] $websiteName,
+        [string] $startStopWebsiteName,
+        [string] $physicalPath,
+        [string] $physicalPathAuth,
+        [string] $physicalPathAuthUserName,
+        [string] $physicalPathAuthUserPassword ,
+
+        [string] $addBinding,
+        [string] $protocol,
+        [string] $ipAddress,
+        [string] $port,
+        [string] $serverNameIndication ,
+        [string] $hostNameWithOutSNI,
+        [string] $hostNameWithHttp,
+        [string] $hostNameWithSNI,
+        [string] $sslCertThumbPrint,
+        [string] $bindings,
+
+        [string] $createOrUpdateAppPool,
+        [string] $appPoolName,
+        [string] $dotNetVersion,
+        [string] $pipeLineMode,
+        [string] $appPoolIdentity,
+        [string] $appPoolUsername,
+        [string] $appPoolPassword,
+
+        [string] $configureAuthentication,
+        [string] $anonymousAuthentication,
+        [string] $basicAuthentication,
+        [string] $windowsAuthentication,
+
+        [string] $appCmdCommands
+    )
+
+    switch ($actionIISWebsite) 
+    {
+        "CreateOrUpdateWebsite" 
+        {
+            Repair-Inputs -siteName ([ref]$websiteName) -physicalPath ([ref]$physicalPath)  -poolName ([ref]$appPoolName) -physicalPathAuthuser ([ref]$physicalPathAuthUserName) -appPoolUser ([ref]$appPoolUsername)
+            
+            if($addBinding -eq "true") 
+            {
+                if([string]::IsNullOrWhiteSpace($bindings)) {
+                    $bindingsArray = @(@{
+                        protocol = $protocol.Trim();
+                        ipAddress = $ipAddress.Trim();
+                        port = $port.Trim();
+                        sniFlag = $serverNameIndication;
+                        sslThumbprint = Test-SSLCertificateThumbprint -sslCertThumbPrint $sslCertThumbPrint -ipAddress $ipAddress -protocol $protocol -port $port ;
+                        hostname = Get-Hostname -port $port -hostNameWithSNI $hostNameWithSNI -hostNameWithHttp $hostNameWithHttp -hostNameWithOutSNI $hostNameWithOutSNI -sni $serverNameIndication ;
+                    })
+                }
+                else {
+                    $bindingsArray = Validate-Bindings -bindings $bindings
+                }
+            }
+
+            $bindingsJson = $bindingsArray | ConvertTo-Json
```

Comment:
Same comment as above for ConvertFrom-Json.

Reply:
Same reply as above for ConvertFrom-Json.